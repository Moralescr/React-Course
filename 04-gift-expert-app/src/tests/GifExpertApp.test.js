import React from 'react';
import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en componente GifExpertApp', () => {

    test('Deberia de mostrar el contenido', () => {

        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
        
    })

    test('deberia de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch', 'Naruto'];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categories }/>);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    })

})