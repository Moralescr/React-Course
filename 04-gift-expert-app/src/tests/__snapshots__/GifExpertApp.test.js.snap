// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifExpertApp Deberia de mostrar el contenido 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente GifExpertApp deberia de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
  </ol>
</Fragment>
`;
